### PR TITLE
Fixes #414

### LINKED ISSUE #414: JSON Serializing Fails on Decimal
The default JSON serializer is not able to serialize Decimal objects.
Exact traceback from a use case:

```
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/plotly/plotly/plotly.py", line 241, in plot
    res = _send_to_plotly(figure, **plot_options)
  File "/Library/Python/2.7/site-packages/plotly/plotly/plotly.py", line 1374, in _send_to_plotly
    cls=utils.PlotlyJSONEncoder)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/Library/Python/2.7/site-packages/plotly/utils.py", line 145, in encode
    encoded_o = super(PlotlyJSONEncoder, self).encode(o)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
  File "/Library/Python/2.7/site-packages/plotly/utils.py", line 211, in default
    return json.JSONEncoder.default(self, obj)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: Decimal('-94.6285553000') is not JSON serializable
```

### PR DESCRIPTION
**Pull request by tarzzz**
Fixes #414 and #384

### CODE REVIEW

**Comment by tarzzz** [comment]
@cldougl Please review. I'll add some test cases as well..

**Comment by tarzzz** [comment]
Yep, it works for the example. Added tests as well.. :)

**Comment by tarzzz** [comment]
@yankev : Getting 502, Bad gateway error on un-related tests (unrelated to this PR).  :/

**Comment by theengineear** [comment]
@tarzzz any update on this? seems like it'd be helpful :)

**Comment by tarzzz** [comment]
@theengineear It is good to go in (I almost forgot I had made this PR) .. a quick 👀 from you or @cldougl before I merge it in .. 😺

File: plotly/matplotlylib/mplexporter/exporter.py [hunk 1]
```diff
@@ -162,8 +162,8 @@ def crawl_legend(self, ax, legend):
                 if isinstance(child, matplotlib.patches.Patch):
                     self.draw_patch(ax, child, force_trans=ax.transAxes)
                 elif isinstance(child, matplotlib.text.Text):
-                    if not (child is legend.get_children()[-1]
-                            and child.get_text() == 'None'):
+                    if (child is not legend.get_children()[-1]
+                            and child.get_text() != 'None'):
```
Comment (theengineear) [inline plotly/matplotlylib/mplexporter/exporter.py]: Can you comment on this change? It seems unrelated?

File: plotly/utils.py [hunk 2]
```diff
@@ -296,6 +298,13 @@ def encode_as_date(obj):
         else:
             return iso_to_plotly_time_string(time_string)
 
+    @staticmethod
+    def encode_as_decimal(obj):
+        """Attempt to encode decimal by converting it to float"""
+        if isinstance(obj, decimal.Decimal):
+            return float(obj)
+        else:
+            raise NotEncodable
```
Comment (theengineear) [inline plotly/utils.py]: 👌

**Comment by theengineear** [comment]
@tarzzz, the default schema you have is likely quite out of date at this point. Would you mind:

1. removing your commit for the default schema change
2. addressing the change to the matplotlylib code (and possibly reverting that?)
3. rebasing on the most current master
4. updating the default schema to the newest version?

I'm also happy to do this, but I am just a little unsure about the motivation for (2).

File: plotly/matplotlylib/mplexporter/exporter.py (hunk 1 above)
Comment (tarzzz) [inline plotly/matplotlylib/mplexporter/exporter.py]: @theengineear 

This fixes #384 ..
I can made it as a different PR, or we can use this PR to kill two birds with a single bullet.. 😸

**Comment by tarzzz** [comment]
@theengineear : Rebased with master, and updated the plot-schema..

File: plotly/matplotlylib/mplexporter/exporter.py (hunk 1 above)
Comment (theengineear) [inline plotly/matplotlylib/mplexporter/exporter.py]: ah, gotcha. 👍

**Comment by theengineear** [comment]
Let's 💃 !
